crates/holochain/src/sweettest/sweet_conductor.rs

```diff
@@ -732,13 +739,47 @@ impl SweetConductor {
                 // Ensure the space is created for DPKI so the agent db exists
                 c.spaces.get_or_create_space(dpki_dna_hash).unwrap();
             }
-            for env in c.spaces.get_from_spaces(|s| s.p2p_agents_db.clone()) {
-                all.push(env.clone());
+            for dna_hash in c.spaces.get_from_spaces(|s| s.dna_hash.clone()) {
+                let agent_infos = c
+                    .holochain_p2p()
+                    .peer_store((*dna_hash).clone())
+                    .await
+                    .unwrap()
+                    .get_all()
+                    .await
+                    .unwrap();
+
+                match all.entry(dna_hash) {
+                    std::collections::hash_map::Entry::Vacant(e) => {
+                        e.insert(agent_infos.into_iter().collect());
+                    }
+                    std::collections::hash_map::Entry::Occupied(mut e) => {
+                        e.get_mut().extend(agent_infos);
+                    }
+                }
```

Comment:
I know I'm the one always voting for match arms instead of functional... but for some reason I like this one better.

```suggestion
                all.entry(dna_hash).or_default().extend(agent_infos);
```

Reply:
That is nice, applied :)